eternalcore-core/src/main/java/com/eternalcode/core/EternalCoreEnvironment.java

```diff
@@ -0,0 +1,52 @@
+package com.eternalcode.core;
+
+import com.google.common.base.Stopwatch;
+import io.papermc.lib.PaperLib;
+import io.papermc.lib.environments.Environment;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+class EternalCoreEnvironment {
+
+    private final EternalCoreApi eternalCoreApi;
+    private final Logger logger;
+    private final Stopwatch stopwatch = Stopwatch.createStarted();
+
+    EternalCoreEnvironment(EternalCoreApi eternalCoreApi, Logger logger) {
+        this.eternalCoreApi = eternalCoreApi;
+        this.logger = logger;
+        this.checkSoftware();
+    }
+
+    void initialize() {
+        EternalCoreApiProvider.initialize(this.eternalCoreApi);
+
+        long millis = this.stopwatch.elapsed(TimeUnit.MILLISECONDS);
+        this.logger.info("Successfully loaded EternalCore in " + millis + "ms");
```

Comment:
I would move Stopwatch to the initialize method.
```suggestion

    EternalCoreEnvironment(EternalCoreApi eternalCoreApi, Logger logger) {
        this.eternalCoreApi = eternalCoreApi;
        this.logger = logger;
        this.checkSoftware();
    }

    void initialize() {
        Stopwatch stopwatch = Stopwatch.createStarted();
        
        EternalCoreApiProvider.initialize(this.eternalCoreApi);

        long millis = stopwatch.elapsed(TimeUnit.MILLISECONDS);
        this.logger.info("Successfully loaded EternalCore in " + millis + "ms");
```